lib/rust/ensogl/component/grid-view/src/lib.rs

```diff
@@ -472,24 +374,48 @@ impl<E: Entry> Model<E, E::Params> {
             let text_layer = text_layer.as_ref().and_then(|l| l.upgrade());
             self.entry_creation_ctx.create_entry(text_layer.as_ref())
         };
-        let entry_frp = {
+        // We must not emit FRP events when some state is borrowed to avoid double borrows.
+        // So the following code block isolates operations with borrowed fields from emitting
+        // FRP events.
+        let (entry_frp, should_set_location) = {
             let mut visible_entries = self.visible_entries.borrow_mut();
             let mut free_entries = self.free_entries.borrow_mut();
-            let entry = match visible_entries.entry((row, col)) {
-                Occupied(entry) => entry.into_mut(),
+            let (entry, should_set_location) = match visible_entries.entry((row, col)) {
+                Occupied(entry) => (entry.into_mut(), false),
                 Vacant(lack_of_entry) => {
                     let new_entry = free_entries.pop().unwrap_or_else(create_new_entry);
                     set_entry_position(&new_entry, row, col, entry_size, &self.column_widths);
-                    new_entry.entry.frp().set_location((row, col));
                     self.display_object.add_child(&new_entry);
-                    lack_of_entry.insert(new_entry)
+                    (lack_of_entry.insert(new_entry), true)
                 }
             };
-            entry.entry.frp().clone_ref()
+            (entry.entry.frp().clone_ref(), should_set_location)
         };
         let width_offset = self.column_widths.width_diff(col);
         entry_frp.set_size(entry_size + Vector2(width_offset, 0.0));
         entry_frp.set_model(model);
+        if should_set_location {
+            entry_frp.set_location((row, col));
+        }
+    }
+
+    fn update_after_column_resize(
+        &self,
+        resized_column: Col,
+        properties: Properties,
+    ) -> Vec<(Row, Col)> {
+        let to_model_request = self.update_entries_visibility(properties);
+        for ((row, col), visible_entry) in &*self.visible_entries.borrow() {
+            // We are not interested in columns to the left of the resized column.
+            if *col < resized_column {
+                continue;
+            }
```

Comment:
We can also use `filter` method of Iterator. This way we avoid jumps inside the blocks.